kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestFileSystemWriters.java

```diff
@@ -95,6 +99,38 @@ public void testBasicWrite() throws IOException {
     Assert.assertEquals("Should match written records",
         written, Lists.newArrayList((Iterator) init(reader)));
   }
+  
+  @Test
+  public void testWriteWithOldSchema() throws IOException {
+    fsWriter = newWriter(testDirectory, SCHEMA);  
+    init(fsWriter);
+
+    FileStatus[] stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain no visible files", 0, stats.length);
+    stats = fs.listStatus(testDirectory);
+    Assert.assertEquals("Should contain a hidden file", 1, stats.length);
+
+    List<Record> written = Lists.newArrayList();
+    for (long i = 0; i < 10000; i += 1) {
+      Record record = record(i, "test-" + i);
+      fsWriter.write(record);
+      written.add(record);
+    }
+
+    stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain no visible files", 0, stats.length);
+    stats = fs.listStatus(testDirectory);
+    Assert.assertEquals("Should contain a hidden file", 1, stats.length);
+
+    fsWriter.close();
+
+    stats = fs.listStatus(testDirectory, PathFilters.notHidden());
+    Assert.assertEquals("Should contain a visible data file", 1, stats.length);
+
+    DatasetReader<Record> reader = newReader(stats[0].getPath(), TEST_SCHEMA);
```

Comment:
Yeah, you're right in reality it should fail the compatibility check but this test case isn't failing as it is a unit test case and the compatibility check happens much before the control goes to FileSystemWriter. Granted I got it wrong here, instead I should have updated the dataset schema and write with the old schema. Coming to the DatasetReader, I think the writer schema is passed on to the reader schema[1]. However, more clarification on this will help me better understand it.  

[1]: https://github.com/kite-sdk/kite/blob/master/kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DataModelUtil.java#L129